commcare-support-library/src/main/java/org/commcare/commcaresupportlibrary/identity/BiometricUtils.java

```diff
@@ -0,0 +1,60 @@
+package org.commcare.commcaresupportlibrary.identity;
+
+import android.util.Base64;
+import android.util.Pair;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import java.lang.reflect.Type;
+
+public class BiometricUtils {
+
+    /**
+     * This method converts the biometric template into a Base64 encoded String to send back to
+     * CommCare, as part of the callout response
+     *
+     * @param biometricIdentifier indicates the type of biometric element captured
+     * @param template inputs the biometric template
+     * @return Base64 encoded string
+     */
+    public static String convertTemplateToBase64String(BiometricIdentifier biometricIdentifier,
+                                                       byte[] template) {
+        // In order to reduce the size of the outpput, the byte array is converted to Base64 before
+        // the instantiation of the pair object
+        Pair<Integer, String> templatePairBase64Encoded = new Pair<>(biometricIdentifier.ordinal(),
+                Base64.encodeToString(template, Base64.DEFAULT));
+        String templatesInJson = new Gson().toJson(templatePairBase64Encoded);
+        return Base64.encodeToString(templatesInJson.getBytes(), Base64.DEFAULT);
```

Comment:
Here's the idea is to bundle the `BiometricIdentifier` with the `template`. It is redundant as we use the `callout response key` to identify the type of biometric but I think it could be useful to have a second layer of identification. For instance, the app builder can make a poor choice of case property name and have difficulties later to know if the template is of a `finger` or a `face`. We could also use the position `0` of the array for this. Any thoughts?

Reply:
> For instance, the app builder can make a poor choice of case property name and have difficulties later to know if the template is of a finger or a face

I don't think its a valid concern. App builders can name the properties of other fields poorly too (First Name to Last Name for ex) and we should not take into account such mistakes to make decisions at platform level. I would rather rgo for simplicity here and just store the template directly in the case property.